app/Http/Traits/CollectionsV2Helpers.php

```diff
@@ -138,13 +138,17 @@ private function checkDatasets(int $collectionId, array $inDatasets, int $userId
             $datasetVersions = DatasetVersion::where('dataset_id', (int) $dataset['id'])->select('id')->get()->toArray();
 
             $datasetVersionIds = convertArrayToArrayWithKeyName($datasetVersions, 'id');
-            $commonDatasetVersionIds = array_intersect($collectionHastDatasetVersionIds, $datasetVersionIds);
+            // This is the intersect of all versions of _this_ dataset, and all the versions in CollectionHasDatasetVersion for this Collection.
+            $commonDatasetVersionIds = array_intersect($collectionHasDatasetVersionIds, $datasetVersionIds);
 
+            // If the supplied datasets are not in the existing set, then add them
             if (count($commonDatasetVersionIds) === 0) {
                 $this->addCollectionHasDatasetVersion($collectionId, $dataset, $datasetVersionLatestId, $userId);
                 continue;
             }
 
+            // else, if the latest version isn't in there, then add it, and remove all the previously existing dataset versions that were linked
+            // - this simply has the effect of updating the link to an already-linked dataset to its latest version id
```

Comment:
can you add less comment or shorter ?!?

Reply:
The aim of these comments is to make the dense logic a little easier to understand when reading the code. Not sure I can make it much shorter while making it legible, and I think clarity is preferable to brevity.

Reply:
the code should speak for itself